internal/provider/kubernetes/helpers.go

```diff
@@ -197,12 +197,12 @@ func refsSecret(ref *gwapiv1b1.SecretObjectReference) bool {
 }
 
 func infraServiceName(gateway *gwapiv1b1.Gateway) string {
-	infraName := utils.GetHashedName(fmt.Sprintf("%s-%s", gateway.Namespace, gateway.Name))
+	infraName := utils.GetHashedName(fmt.Sprintf("%s/%s", gateway.Namespace, gateway.Name))
```

Comment:
nice find :) 
I understand this issue, which stems from https://github.com/envoyproxy/gateway/blob/c0a619ca85c916658424d6b74870744bf14a4f1d/internal/gatewayapi/helpers.go#L412 & https://github.com/envoyproxy/gateway/blob/c0a619ca85c916658424d6b74870744bf14a4f1d/internal/infrastructure/kubernetes/proxy/resource.go#L54

can we add a comment in here explaining why its this way or should we change 

or instead of finding the deployment by its name which is done here 
https://github.com/envoyproxy/gateway/blob/c0a619ca85c916658424d6b74870744bf14a4f1d/internal/provider/kubernetes/predicates.go#L270

should we find the deployment based on the gateway labels https://github.com/envoyproxy/gateway/blob/c0a619ca85c916658424d6b74870744bf14a4f1d/internal/gatewayapi/helpers.go#L292 which makes more sense imo

Reply:
i think the way we had is great.

and i have added some comments about `func GetHashedName` in `internal/provider/utils/utils.go#L26`: this function takes input which should be formatted as `{Namespace}/{ResourceName}`